core/src/main/java/org/mule/runtime/core/policy/ParameterizedPolicy.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.policy;
+
+import static java.util.Collections.emptyMap;
+import org.mule.runtime.api.metadata.TypedValue;
+
+import java.util.Map;
+
+/**
+ * This class represents a policy injection.
+ * <p>
+ * It contains the set of parameters configured to apply the policy and the chain of
+ * {@link org.mule.runtime.core.api.processor.Processor}s to be applied.
+ */
+public class ParameterizedPolicy {
```

Comment:
I think there's a naming issue here. I will rename this class to Policy since the sum of the parameters + the policy chain represents a policy and rename the Policy to PolicyInstance. Eventually there will be a factory for PolicyInstance which will be called PolicyFactory.